aten/src/ATen/native/Pow.cpp

```diff
@@ -11,7 +11,8 @@ DEFINE_DISPATCH(pow_tensor_tensor_stub);
 DEFINE_DISPATCH(pow_tensor_scalar_stub);
 
 Tensor& pow_out(Tensor& result, const Tensor& base, const Tensor& exp) {
-  if (exp.dim() == 0 && base.dim() != 0) {
+  if (exp.dim() == 0 && exp.device().type() == DeviceType::CPU
```

Comment:
We can't cast cuda tensor with `dim=0` into its unwrapped form (because it lives in the different memory space), so checking exp device type is reasonable. 
Checking `base` device type is not strictly necessary (you could take tensor-scalar path even for cpu bases) but does not hurt.

Reply:
yes. what I meant is create implicit type cast for `tensor(dim==0)` and it should have a CPU impl correctly, and probably raise exception for CUDA impl (HIP impl). this way we dont need to address if (exp.dim() == 0) separately. 
But this goes beyond the scope of this PR IMO.